pcdsdevices/piezo.py

```diff
@@ -0,0 +1,97 @@
+"""
+Module for piezo drivers.
+"""
+
+import logging
+
+from ophyd.device import Component as Cpt
+from ophyd.device import Device
+from ophyd.signal import EpicsSignal, EpicsSignalRO
+
+from pcdsdevices.variety import set_metadata
+
+logger = logging.getLogger(__name__)
+
+
+class SliceDhvChannel(Device):
+    """
+    Class for controlling a single channel from a Vescent Photonics Slice-DHV
+    piezo driver channel PVs.
+    """
+
+    voltage_limit = Cpt(EpicsSignal, ':VLIM_GET', write_pv=':VLIM_SET',
+                        kind='config')
+
+    control_settings = Cpt(EpicsSignal, ':CTL_SETTINGS_GET',
+                           write_pv=':CTL_SETTINGS_SET', kind='config')
+
+    bias_voltage = Cpt(EpicsSignal, ':VMEAS_GET', write_pv=':VBIAS_SET',
+                       kind='normal')
+
+    sweep_voltage = Cpt(EpicsSignal, ':VSWEEP_GET', write_pv=':VSWEEP_SET',
+                        kind='config')
+
+    sweep_freq = Cpt(EpicsSignal, ':FSWEEP_GET', write_pv=':FSWEEP_SET',
+                     kind='config')
+
+    sweep_mode = Cpt(EpicsSignal, ':SWEEP_MODE_GET',
+                     write_pv=':SWEEP_MODE_SET', kind='config')
+
+    error_status = Cpt(EpicsSignalRO, ':ERROR', kind='config')
+
+    trigger_in = Cpt(EpicsSignal, ':TRIGIN_STATUS_GET',
+                     write_pv=':TRIGIN_STATUS_SET', kind='config')
+
+    trigger_out = Cpt(EpicsSignal, ':TRIGOUT_STATUS_GET',
+                      write_pv=':TRIGOUT_STATUS_SET', kind='config')
+
+    modulation = Cpt(EpicsSignal, ':MODULATION_GET',
+                     write_pv=':MODULATION_SET', kind='config')
+
+    output_mode = Cpt(EpicsSignal, ':OUTPUT_MODE_GET',
+                      write_pv=':OUTPUT_MODE_SET', kind='config')
+
+
+class SliceDhvController(Device):
+    """
+    Class for the Vescent Photonics Slice-DHV piezo driver controller PVs.
+    """
+
+    _command = Cpt(EpicsSignal, ':CMD', kind='omitted')
+
+    _reply = Cpt(EpicsSignalRO, ':REPLY', kind='omitted')
+
+    version = Cpt(EpicsSignalRO, ':VERSION', kind='omitted')
+
+    reset_ch1 = Cpt(EpicsSignal, ':FACTORY_RESET_CH1', kind='omitted')
+    set_metadata(reset_ch1, dict(variety='command-proc', value=1))
+
+    reset_ch2 = Cpt(EpicsSignal, ':FACTORY_RESET_CH2', kind='omitted')
+    set_metadata(reset_ch2, dict(variety='command-proc', value=1))
+
+    save_config = Cpt(EpicsSignal, ':SAVE_CONFIG', kind='config')
+    set_metadata(save_config, dict(variety='command-proc', value=1))
+
+    save_status = Cpt(EpicsSignalRO, ':SAVE_STATUS', kind='config')
+
+    manufacturer = Cpt(EpicsSignalRO, ':MANUFACTURER', kind='omitted')
+
+    model = Cpt(EpicsSignalRO, ':MODEL', kind='omitted')
+
+    serial = Cpt(EpicsSignalRO, ':SERIAL', kind='omitted')
+
+    ctl_fw = Cpt(EpicsSignalRO, ':CTL_FW', kind='omitted')
+
+    dhv_fw = Cpt(EpicsSignalRO, ':DHV_FW', kind='omitted')
+
+
+class SliceDhv(Device):
```

Comment:
One minor thing - tab completion filtering and such won't be available on bare `ophyd.Device` classes. We generally mix in [`BaseInterface`](https://github.com/pcdshub/pcdsdevices/blob/8abc4868395ef923dba8bec68760823744cf0394/pcdsdevices/interface.py#L172) to get status printing helpers, tab filtering, etc.

If this is solely for typhos-based usage, it's probably OK to skip that for now.

Reply:
Does it hurt to add BaseInterface? If not, might as well just in case we end up using them in hutch-python at some point.

Reply:
I will certainly want to use this in hutch python (part of why I wrote this :) ). I'll add that in.